https://issues.redhat.com/browse/ACM-14599 (#7073)
open-cluster-management/rhacm-docs: governance/deploy_pol_ext_tools.adoc

[#deploy-pol-ext-tools]
= Deploying policies with external tools

To deploy `CertificatePolicy`, `ConfigurationPolicy`, `OperatorPolicy` resources, and Gatekeeper constraints directly to your managed cluster, use an external tool such as {gitops}.

*Required access:* Cluster administrator

[#pol-ext-tools-prereq]
== Prerequisites

> * Your `CertificatePolicy`, `ConfigurationPolicy`, `OperatorPolicy` policies must in the `open-cluster-management-policies` namespace or managed cluster namespace. Policies in other namespaces are ignored and do not receive a status. 

Reviewer: ```suggestion
* Your `CertificatePolicy`, `ConfigurationPolicy`, `OperatorPolicy` policies must in the `open-cluster-management-policies` namespace or the managed cluster namespace. Policies in other namespaces are ignored and do not receive a status. 
```

Author: ```suggestion
* Your `CertificatePolicy`, `ConfigurationPolicy`, `OperatorPolicy` policies must be in the `open-cluster-management-policies` namespace or the managed cluster namespace. Policies in other namespaces are ignored and do not receive a status. 
```

> * You must configure {acm} policy health checks if you are using a {gitops} version with an Argo CD version earlier than 2.13. See link:../gitops/gitops_manage_policy_def.adoc#config-gitops-healthcheck[Configuring policy health checks in {gitops-short}.

Reviewer: ```suggestion
* You must configure {acm} policy health checks if you are using a {gitops} version with an Argo CD version earlier than 2.13. See link:../gitops/gitops_manage_policy_def.adoc#config-gitops-healthcheck[Configuring policy health checks in {gitops-short}].
* The {gitops} service account must have permission to manage {acm} policies. See link:../gitops_manage_policy_def.adoc#create-clusterrole-gitops[Creating a ClusterRole resource for {gitops}].
```

Author: Great catch!

> * You must configure {acm} policy health checks if you are using a {gitops} version earlier than 2.13. You are not required to configure {acm} policy health checks for {gitops-short} version later than 2.13. See link:../gitops/gitops_manage_policy_def.adoc#config-gitops-healthcheck[Configuring policy health checks in {gitops-short}.

Reviewer: It's confusing but the OpenShift GitOps operator doesn't use the same version as the Argo CD it deploys.
```suggestion
* You must configure {acm} policy health checks if you are using a {gitops} version with an Argo CD version that is older than 2.13. You are not required to configure {acm} policy health checks for Argo CD versions later than 2.13. See link:../gitops/gitops_manage_policy_def.adoc#config-gitops-healthcheck[Configuring policy health checks in {gitops-short}.
```

Author: When you say older, I understand that to mean an earlier or previous version
```suggestion
* You must configure {acm} policy health checks if you are using a {gitops} version with an Argo CD version earlier than 2.13. You are not required to configure {acm} policy health checks for {gitops-short} version later than 2.13. See link:../gitops/gitops_manage_policy_def.adoc#config-gitops-healthcheck[Configuring policy health checks in {gitops-short}.
```

> * You must configure {acm} policy health checks if you are using a {gitops} version with an Argo CD version earlier than 2.13. You are not required to configure {acm} policy health checks for {gitops-short} version later than 2.13. See link:../gitops/gitops_manage_policy_def.adoc#config-gitops-healthcheck[Configuring policy health checks in {gitops-short}.

Reviewer: ```suggestion
* You must configure {acm} policy health checks if you are using a {gitops} version with an Argo CD version earlier than 2.13. Newer versions will have this included by default. See link:../gitops/gitops_manage_policy_def.adoc#config-gitops-healthcheck[Configuring policy health checks in {gitops-short}.
```

Author: I actually think we should not mention that the configuration is not required for newer versions. It seems like an early announcement from my perspective. What if a bug occurs in a {gitops} version with a newer Argo CD version, where the {acm} configuration is not included?

Reviewer: @dockerymick that's fair. We can update the doc when there is a new GitOps version with the ACM health checks to clarify this.

So I'm good if you want to always say the health check configuration is needed.

[#deploy-pol-gitops]
== Deploying policies with {giops-short} 

> You can deploy policies with {gitops-short} and view the policies from the _Discovered policies_ table of the Governance dashboard. The policies are grouped by the policy `name` and `kind` parameters. 

Reviewer: ```suggestion
You can deploy policies with {gitops-short} and view the policies from the _Discovered policies_ table of the Governance dashboard. The policies are grouped by the policy `name` and `kind` fields. 
```

> You can deploy policies with {gitops-short} from the _Discovered policies_ table of the Governance dashboard. The policies are grouped by the policy `name` and `kind` parameters. 

Reviewer: The policies are not deployed through the discovered policies, but the discovered policies table can be used to view the policies. OpenShift GitOps or some other tool is the tool that deploys it.

> You can deploy policies with {gitops-short} from the _Discovered policies_ table of the Governance dashboard. The policies are grouped by the policy `name` and `kind` parameters. 

Author: ```suggestion
You can deploy policies with {gitops-short} and view the policies from the _Discovered policies_ table of the Governance dashboard. The policies are grouped by the policy `name` and `kind` parameters. 
```

> //could we add steps for deploying the configuration policy example that you mention in the google doc? 

Author: @mprahl

Reviewer: @dockerymick could you please provide a bit more information? Are you asking if we can provide an end to end example of OpenShift GitOps deploying a ConfigurationPolicy?

Author: Yes, I am asking to add a end to end example of OpenShift GitOps deploying a ConfigurationPolicy

Reviewer: @dockerymick I don't think it's necessary. It's not really any different than using OpenShift GitOps to deploy anything else so I think we can leave it.

Author: Ok. Based on the title that we agreed to I assumed that this would be a task-focused file. To make it more like a concept topic, what do you think of the following title suggestion: "Policy deployment by using external tools"

If you agree with this update, then we do not need the prerequisites section.

Reviewer: @dockerymick that title works for me. Thanks!
